lib_bgp_data/roas_parser/roas_automator.py

```diff
@@ -0,0 +1,191 @@
+                            
+                    # If after three attempts the backups are unsuccessful,
+                    # then just insert the new roas and don't backup.
+                    else:
+                        utils.clean_paths([self.path, self.csv_dir])
+                        utils.rows_to_db(roas, _csv_dir,
+                                         ROAs_Table, clear_table=False)
+                        table.create_index()
+
+                        # Raise a backup error
+                        raise BackupError()
+
+                except subprocess.CalledProcessError as e:
+                    time = utils.now()
+                    subject = f"Roas Backup Error at {time}"
+                    body = ("There was an error backing up the roas table. "
+                            "Below are the outputs\n"
+                            f"STD_OUT Message: {e.stdout}\n"
+                            f"STD_ERR Message: {e.stderr}")
+                    utils.send_email(subject, body)
+
+                except BackupError:
+                    time = utils.now()
+                    subject = f"Roas Backup Error at {time}"
+                    body = ("There was an error backing up the roas table. "
+                            "The backup file was not created correctly, so "
+                            "the previous backup was not overwritten. The "
+                            "pg_dump and pg_restore commands did not provide "
+                            "an error.")
+                    utils.send_email(subject, body)
+
+########################
+### Helper Functions ###
+########################
+
+    def _backup_table(self):
+        """This funtion will backup the table and check the backup"""
+
+        # Get the number of rows in the table
+        with ROAs_Table() as table:
+            count = table.get_count()
+            
+        # Make a temporary backup
+        cmd = (f"sudo -i -u postgres "
+               f"pg_dump -Fc -t roas bgp > {self.temp_backup}")
+        subprocess.run(cmd, check=True, capture_output=True,
+                       shell=True, text=True)
+
+        # Restore the table from the temporary backup we just made
+        with ROAs_Table() as table:
+            try:
+                table.clear_table()
+                self._restore_table(temp=True)
+                table.create_index()
+            except:
+                # If there are any errors trying to restore the table,
+                # the backup was not successful
+                return False
+
+        # Get the new count of the table after the restore
+        with ROAs_Table() as table:
+            restore_count = table.get_count()
+
+        # Return True if the backup is consistent
+        return count == restore_count
+            
+    def _restore_table(self, temp=False):
+        """This function runs the command to restore the table from the
+        backup"""
+
+        cmd = "sudo -i -u postgres "
```

Comment:
Same as above, it would be nice if you could move this into postgres.py and add a unit test for it. Section should be an input, the base class Parser has a default section that should be used instead of bgp.